### DISCUSSION TITLE
Support formatting by reading from stdin and printing to stdout

### DISCUSSION THREAD

**Discussion by RafaelKr** [PHP-CS-Fixer/PHP-CS-Fixer#8309] [Feature ideas]
We're currently using https://dprint.dev/ as a formatter.
By using plugins it's able to format a lot of different file types. Based on the file extension it's deciding which formatter to use. This has the advantage that we just need to call one command and every supported file in the whole project is formatted. Also in the IDE we just need to configure one command which is called for format-on-save.

Usually dprint passes the contents via stdin, reads the formatted result from stdout and writes that back to the file.

For PHP files we configured https://github.com/dprint/dprint-plugin-exec/ to call PHP-CS-Fixer. But we need to have a wrapper script which writes the stdin to a temporary file, let PHP-CS-Fixer do the formatting, read the file and print it to stdout. First we used PHP-CS-Fixer to directly write to the file, but then the file is written twice which caused issues for the colleague using VS Code.

You can see the wrapper script implementation at https://github.com/dprint/dprint-plugin-exec/issues/26#issuecomment-2498294561

It would be much better (probably faster and definitely easier on the SSD) to be able to use PHP-CS-Fixer via stdin and get the result via stdout.

Additional References:
- https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/issues/2011 (and https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/4320)
- https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/issues/454
- https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/issues/784
- https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/issues/1579
- https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/issues/1675
- https://github.com/dprint/dprint/issues/915
- https://github.com/VincentLanglet/Twig-CS-Fixer/discussions/332 (Same request for Twig-CS-Fixer)

**Comment by keradus** [comment]
converted to issue, happy to have this developed if one likes to see this in

#9189
